### PR TITLE
HCNY

### PR DESCRIPTION
**Pull request by gitgood-ENT**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -10,42 +12,95 @@
...
 
+// Describe a function called "descriptor" that takes in an array of objects and returns an array with a sentence about each person with their name capitalized
+describe("descriptor", () => {
+  let people = [
+    { name: "ford prefect", occupation: "a hitchhiker" },
+    { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+    { name: "arthur dent", occupation: "a radio employee" }
+  ]
+  let expectedOutput = ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+  it("functionFunction", () => {
```
Comment (elysemontano) [inline code-challenges.test.js]: The "it" block on your tests should be more descriptive than "functionFunction".  For ex: it("takes an array of objects and returns an array with names capitalized in a sentence", etc..).

File: code-challenges.test.js [hunk 2]
```diff
@@ -10,42 +12,95 @@
...
+})
+
+// RED
+
 // b) Create the function that makes the test pass.
 
+// Create a function that takes in an array and filters out the non-numbers and then uses map to return the remainder of the numbers
+const remainders = (arr) => {
+  return arr.filter(value => typeof value === 'number').map(value => value % 3)
+}
```
Comment (elysemontano) [inline code-challenges.test.js:79]: Great work here!  Clean and concise.  Well done!

File: code-challenges.test.js [hunk 3]
```diff
@@ -10,42 +12,95 @@
...
+  })
+})
+// RED
 
 // b) Create the function that makes the test pass.
+
+// Create a function that takes in an array, cubes each of the numbers using map, and then adds them
+const cubeAndSum = (arr) => {
+  return arr.map(value => value ** 3).reduce((a, b) => (a + b))
+}
```
Comment (elysemontano) [inline code-challenges.test.js:104]: Perfect!  Logic is great on all of these problems.

File: interview-questions.md [hunk 4]
```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: add the foreign key through a migration. The foreign key name is foreign_key, or something els that I decide. The key would be on the student model and be referenced in the cohort model
```
Comment (elysemontano) [inline interview-questions.md:10]: Awesome answer!  Just as an FYI, Rails is very particular about the naming convention.  The foreign key name would have to be the model it belongs to followed by id.  For ex: cohort_id would be the foreign key in this instance.  Any variation from that and rails might not read it properly.

File: interview-questions.md [hunk 5]
```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+As a User: I can see colummns for To do, Date started, Date Due, complete?, Date completed 
+As a User: I can add custom colummns to the list
+As a User: I can add items to the list
+As a User: I can edit the information in each box
+As a User: I can see a desciption of the list
+As a User: I can select multipe items to be deleted/updated
+As a User: I can see different colors based on importance/priority levels
+As a User: I can change the size of the table 
+As a User: I can select different table styles
+As a User: I can change the orderof list items, or sort them based on due date or priority level
```
Comment (elysemontano) [inline interview-questions.md:68]: Great work!  Sounds like this is a very thorough to-do list!

File: rails-commenting.rb [hunk 6]
```diff
@@ -55,14 +62,17 @@ def destroy
...
   # ---9)
+  # This is where the private methods are stored. Private methods can only be accessed by the defining class, or by methods within the class that access it.
   private
   def blog_post_params
-    # ---10)
+    # ---10
+    # These are strong params and will pass an error to the user if they are not satisfied when attempting to create or update data, or whatever method calls on this private method
     params.require(:blog_post).permit(:title, :content)
   end
 end
```
Comment (elysemontano) [inline rails-commenting.rb:78]: Great work on this section.  Really well thought out answers!

**Review by elysemontano — CHANGES_REQUESTED** [review]
Overall great job Noah!  You crushed this assessment.  Happy to approve this pull request once your tests are more descriptive.  Other than that, everything looks awesome!

File: code-challenges.test.js (hunk 1 above)
Comment (gitgood-ENT) [reply code-challenges.test.js]: Should be Fixed!

**Review by elysemontano — CHANGES_REQUESTED** [review]
First test looks great.  The second and third test still needs to be more descriptive though.

File: code-challenges.test.js (hunk 1 above)
Comment (gitgood-ENT) [reply code-challenges.test.js]: The rest are fixed

File: code-challenges.test.js (hunk 2 above)
Comment (gitgood-ENT) [reply code-challenges.test.js:79]: Should be completely fixed now.

**Review by elysemontano — APPROVED** [review]
Looks good!

**Pull request merged**